### DISCUSSION TITLE
Questions for v2.5.0 release note

### DISCUSSION THREAD

**Discussion by hueiyuan** [Altinity/clickhouse-backup#905] [Q&A] [answered]
Hi, all
We found that [v2.5.0](https://github.com/Altinity/clickhouse-backup/releases/tag/v2.5.0) has been released. And some of list we want to confirm whether have side-effect.
* `added clickhouse/clickhouse-server:24.3 to CI/CD` --> This means do we need to upgrade clickhouse server to 24.3 version?
*  `added --delete-source parameter for upload and create_remote commands to explicitly delete local backup during upload` --> If we execute `watch` command, so does this version explicitly and automatically delete local backup during upload?
* After v2.5.0, Can the total create_remote and upload s3 performance be better?

Hope related contributors can assist to answer these questions, thank you.

**Comment by Slach** [comment, marked as answer]
> added clickhouse/clickhouse-server:24.3 to CI/CD --> This means do we need to upgrade clickhouse server to 24.3 version?

no it means clickhouse-backup is checked to compatibility with 24.3 version

>  If we execute watch command, so does this version explicitly and automatically delete local backup during upload?

no, `watch` command executes sequences `create_remote` + `delete local` so local backup will delete AFTER upload

and maybe we need to 

`--delete-source` allows us to free disk space during massive clickhouse data mutations 
data parts in clickhouse are immutable, 
`create` command creates hard-links, so during mutations clickhouse create new data parts and this is could allocates disk space

> After v2.5.0, Can the total create_remote and upload s3 performance be better?

Better use 2.5.5 and speed depends on `upload_concurrency` and your available CPU, Network, RAM, disk I/O resources, 2.4.x and 2.5.x is similar

2.5.5 shall be much better for backup object disks (s3, azure)
